services/scheduler/src/SchedulerApp.js

```diff
@@ -1,42 +1,38 @@
 const express = require('express');
 const {
     QueueCreator,
-    App,
-    K8sService,
-    AMQPService
+    App
 } = require('backend-commons-lib');
 const { Scheduler } = require('@openintegrationhub/scheduler');
 const FlowsDao = require('./FlowsDao');
 const SchedulePublisher = require('./SchedulePublisher');
+const { asValue, asClass } = require('awilix');
 
 class SchedulerApp extends App {
     async _run() {
-        this._amqp = new AMQPService(this);
-        this._k8s = new K8sService(this);
-        await this._amqp.start();
-        await this._k8s.start();
-        this._initHealthcheckApi(this.getConfig().get('LISTEN_PORT'));
-        this._channel = await this._amqp.getConnection().createChannel();
-        this._queueCreator = new QueueCreator(this._channel);
+        const container = this.getContainer();
+        const config = container.resolve('config');
+        const amqp = container.resolve('amqp');
+        const k8s = container.resolve('k8s');
+        await amqp.start();
+        await k8s.start();
+        this._initHealthcheckApi(config.get('LISTEN_PORT'));
+        const channel = await amqp.getConnection().createChannel();
+        const queueCreator = new QueueCreator(channel);
+
+        container.register({
+            crdClient: asValue(k8s.getCRDClient()),
+            channel: asValue(channel),
+            queueCreator: asValue(queueCreator),
+            flowsDao: asClass(FlowsDao),
+            schedulePublisher: asClass(SchedulePublisher),
```

Comment:
may be  asClass().singleton?
otherwise it will create new instance of published for ever call of container.resolve('scheduler')
or I'm wrong?